### PR TITLE
[bitnami/harbor] Fix and standardize internalTLS

### LINKED ISSUE #3489: [bitnami/harbor] Several containers fail to start when internalTLS is enabled
**Which chart**:
Harbor v7.1.0

**Describe the bug**
When enabling internalTLS, several containers fail to start.

The chartmuseum container enters a crashloop with this error: FATAL	open /bitnami/certs/server.crt: no such file or directory
The secret being mounted for ChartMuseum is created, but has no server.crt key.

The jobservice container enters a crashloop with this error: Failed on load rest config err:Get https://harbor-core:443/api/internal/configurations: x509: certificate signed by unknown authority, url:https://harbor-core:443/api/internal/configurations
It does not trust the automatically generated certificate that the core is using.

The harbor-portal container enters a crashloop, hanging at ** Starting NGINX **

The trivy container never starts, hanging at: {"addr":":8443","certificate":"/bitnami/certs/tls.crt","clientCAs":"/bitnami/certs/ca.crt","key":"/bitnami/certs/tls.key","level":"debug","msg":"Starting API server with TLS","time":"2020-08-22T10:32:53Z"}

**To Reproduce**
1. Install harbor with internalTLS enabled.

**Expected behavior**
All containers will start normally when using internalTLS

### PR DESCRIPTION
**Pull request by joancafom**
Signed-off-by: joancafom <jcarmona@bitnami.com>

**Description of the change**
This PR is aim at fixing the issues that caused the Chart to fail when enabling `internalTLS`,  as detailed in #3489 .

The internalTLS makes the internal components to communicate with each other securely. The components that support this feature are:

- Chartmusem
- Clair
- Core
- JobService
- Portal
- Registry
- Trivy

We need to update their respective docker images as in order to support this feature, the internal CA issuing the certificates needs to be trusted. 

Also, paths regarding internal TLS certificates have been standardized to match upstream, as well as the rest of the paths this chart uses do. This new paths are:

-  `/etc/harbor/{tls, ssl}/` --> Certificates and Keys for internal TLS can be found here
-  `/harbor_cust_cert/` --> CAs that need to be trusted should be placed here. We should adapt the containers to make sure they take them into account.
  

**Benefits**
Internal components can make use of a secure connection to communicate. New directory that allows the installation of custom CAs that can be used in #3486.

**Possible drawbacks**
Not really, this feature is not working at the moment.

**Applicable issues**

  - fixes #3489

**Additional information**

**Checklist** 
- [x] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/).
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[bitnami/chart]`)

:warning: Keep in mind that if you want to make changes to the kubeapps chart, please implement them in the [kubeapps repository](https://github.com/kubeapps/kubeapps/tree/master/chart/kubeapps). This is only a synchronized mirror.

### CODE REVIEW

File: bitnami/harbor/templates/core/core-cm.yaml
Comment (juan131) [inline bitnami/harbor/templates/core/core-cm.yaml:20]: Why do we want to change the default run mode?

**Review by juan131 — COMMENTED** [review]
Thanks for the changes!! Is it WIP yet? I don't see anything that worries me but please test the upgrade path with these changes, we should be extremely careful to avoid introducing backwards incompatibilities.

File: bitnami/harbor/templates/core/core-cm.yaml
Comment (joancafom) [reply bitnami/harbor/templates/core/core-cm.yaml:20]: I just change this in order to match the upstream repo (which used to use `runmode=dev` as well):

https://github.com/goharbor/harbor-helm/blob/f5767efaa35f835934ce060aa467b3a00af82837/templates/core/core-cm.yaml#L8

**Comment by joancafom** [comment]
Yes @juan131 , there are a couple of things I am still working on:

1. Check for README in case we need to update
2. Make sure upgrading works as expected, so we can go with a Minor instead of a Major
3. Implement changes in the aforementioned containers and wait for their release. Without the new images this PR would not work.

**Comment by juan131** [comment]
Please ping me once I should review the changes again

**Comment by joancafom** [comment]
Ping @juan131 

This is ready to be reviewed again 😁 ! Tested upgrading in Minikube cluster and did not find issues, hence going for a Minor.

**Review by juan131 — APPROVED** [review]
LGTM

**Pull request merged**
